BEIMA.Backend.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,227 @@
+using System.IO;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System;
+using MongoDB.Bson;
+
+namespace BEIMA.Backend.Test
+{
+    [TestFixture]
+    public class MongoConnectorTest
+    {
+        //private MongoConnector? mongo;
+        class LocalSettings
+        {
+            public bool IsEncrypted { get; set; }
+            public Dictionary<string, string> Values { get; set; }
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            //For local testing, the try block will be used to instantiate local variables.
+            //For cloud testing, this will throw an exception, and it will use the env variables
+            //defined in beima.yml.
+            try
+            {
+                string[] paths = { "..", "..", "..", "..", "BEIMA.Backend", "local.settings.json" };
+                string combinedPath = Path.Combine(paths);
+                string fullPath = Path.GetFullPath(combinedPath);
+                var settings = JsonConvert.DeserializeObject<LocalSettings>(
+                    File.ReadAllText(fullPath));
+
+                foreach (var setting in settings.Values)
+                {
+                    Environment.SetEnvironmentVariable(setting.Key, setting.Value);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
+            }
+            
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+        }
+
+        [Test]
+        public void InitialState_Action_ExpectedResult()
+        {
+            Assert.IsTrue(true);
+        }
+
+        [Test]
+        public void ConnectorNotCreated_CallConstructorFiveTimes_FiveInstancesAreEqual()
+        {
+            //Tests to make sure singleton is working
+            var connector1 = MongoConnector.Instance;
+            var connector2 = MongoConnector.Instance;
+            var connector3 = MongoConnector.Instance;
+            var connector4 = MongoConnector.Instance;
+            var connector5 = MongoConnector.Instance;
+            Assert.IsTrue(connector1 == connector2);
+            Assert.IsTrue(connector1 == connector3);
+            Assert.IsTrue(connector1 == connector4);
+            Assert.IsTrue(connector1 == connector5);
+        }
+
+        [Test]
+        public void ConnectorNotCreated_CallConstructorOnce_InstanceConnectedToCorrectEnvironment()
+        {
+            var mongo = MongoConnector.Instance;
+            if (Environment.GetEnvironmentVariable("CurrentEnv") == "dev-local")
+            {
+                Assert.IsTrue(mongo.CurrentServerType == ServerType.Local);
+            }
+            else
+            {
+                Assert.IsTrue(mongo.CurrentServerType == ServerType.Cloud);
+            }
```

Comment:
We should make this test only expect one type, that way if it fails we know exactly what environment it was set in. Since the MongoConnector is a singleton, I think we can create another test class that tests when this environment variable is not "dev-local"

Reply:
We could even have another test class that doesn't initialize the client to verify the correct error message is given

Reply:
This test has been removed after an offline conversation. This should be an integration or functional test, it does not make sense for it to be a unit test. afb7770